site/src/components/FavoriteButton.js

```diff
@@ -30,79 +59,49 @@ const DeleteFavorite = gql`
   }
 `
 
-const GetFavoriteId = gql`
-  query favoriteIdQuery($userId: String, $recordId: String!) {
-    favorites(
-      where: { user_id: { _eq: $userId }, record_id: { _eq: $recordId } }
-    ) {
-      id
-    }
-  }
-`
-
-const GetCount = gql`
-  query countQuery($recordId: String!) {
-    favorites_aggregate(where: { record_id: { _eq: $recordId } }) {
-      aggregate {
-        count
-      }
-    }
-  }
-`
-
 export default function FavoriteButton({
   recordId,
   className,
-  favoriteId: existingFavoriteId,
+  count: propCount,
+  favoriteId: propFavoriteId,
 }) {
-  const [favoriteId, setFavoriteId] = useState(existingFavoriteId)
-  const [count, setCount] = useState(0)
-  const [userId, setUserId] = useState()
-  const { loading: authenticationLoading, user } = useAuth0()
+  const [favoriteId, setFavoriteId] = useState()
+  const [count, setCount] = useState()
+  const favorited = !!favoriteId
 
   useEffect(() => {
-    if (authenticationLoading) return
-    setUserId(user.sub)
-  }, [authenticationLoading])
-
-  useQuery(GetFavoriteId, {
-    onCompleted: data => setFavoriteId(data.favorites[0]?.id),
-    variables: {
-      userId,
-      recordId,
-    },
-  })
-
-  useQuery(GetCount, {
-    onCompleted: data => setCount(data.favorites_aggregate.aggregate.count),
-    variables: { recordId },
-  })
+    setFavoriteId(propFavoriteId)
+    setCount(propCount || 0)
+  }, [propFavoriteId, propCount])
```

Comment:
Just learned that we need to track `props` if they might change, which is the case here as favorites get loaded in.